internal/yml/yml_test/compare_test.go

```diff
@@ -132,3 +134,145 @@ foo:
 		})
 	}
 }
+
+func TestEqualWithExclusion(t *testing.T) {
+	tests := []struct {
+		name         string
+		yaml1        string
+		yaml2        string
+		excludedPath []string
+		expected     bool
+	}{
+		{
+			name: "Identical trees without exclusion",
+			yaml1: `
+foo:
+  bar: baz
+  nested:
+    - item1
+    - item2
+`,
+			yaml2: `
+foo:
+  bar: baz
+  nested:
+    - item1
+    - item2
+`,
+			excludedPath: nil,
+			expected:     true,
+		},
+		{
+			name: "Different trees without exclusion",
+			yaml1: `
+foo:
+  bar: baz
+  nested:
+    - item1
+    - item2
+`,
+			yaml2: `
+foo:
+  bar: bam
+  nested:
+    - item1
+    - item2
+`,
+			excludedPath: nil,
+			expected:     false,
+		},
+		{
+			name: "Identical trees excluding node at root level",
+			yaml1: `
+foo: baz
+bar: qux
+`,
+			yaml2: `
+foo: bam
+bar: qux
+`,
+			excludedPath: []string{"foo"},
+			expected:     true,
+		},
+		{
+			name: "Identical trees excluding nested node",
+			yaml1: `
+foo:
+  bar: baz
+  nested:
+    - item1
+    - item2
+`,
+			yaml2: `
+foo:
+  bar: baz
+  nested:
+    - item3
+    - item4
+`,
+			excludedPath: []string{"foo", "nested"},
+			expected:     true,
+		},
+		{
+			name: "Different trees excluding node at root level",
+			yaml1: `
+foo: baz
+bar: qux
+`,
+			yaml2: `
+foo: bam
+bar: qul
+`,
+			excludedPath: []string{"foo"},
+			expected:     false,
+		},
+		{
+			name: "Different trees excluding nested node",
+			yaml1: `
+foo:
+  bar: baz
+  nested:
+    - item1
+    - item2
+`,
+			yaml2: `
+foo:
+  bar: bak
+  nested:
+    - item3
+    - item4
+`,
+			excludedPath: []string{"foo", "nested"},
+			expected:     false,
+		},
+		{
+			name: "Different trees with non-matching exclusion",
+			yaml1: `
+foo:
+  bar: baz
+`,
+			yaml2: `
+foo:
+  bar: bam
+`,
+			excludedPath: []string{"foo", "nested"},
+			expected:     false,
+		},
+	}
```

Comment:
I would add a test case for the bug (Get ready for more Xmas PLOUP)

```go
{
	name: "ploup!",
	yaml1: `{
		a: {
			b: alpha
		}
	}`,
	yaml2: `{
		a: {
			b: beta
		}
	}`,
	excludedPath: []string{"definitely-not-a", "b"},
	expected:     false,
},
```

You'll notice that the objects are not equal, but since the values that differentiate are at the same level as the expected key and have the same last segment their difference gets ignored